File: code/poptipwindow/src/main/java/com/bingor/poptipwindow/view/OnItemSelectListener.java
package com.bingor.poptipwindow.view;

import com.bingor.poptipwindow.view.wheel.WheelView;

/**
 * Created by HXB on 2018/10/16.
 */
public interface OnItemSelectListener<DataType> {
    /**
     * 滑动选择回调
     *
     * @param index 当前选择项的索引
     */
    <View extends WheelView> void onSelected(View wheelView, int index, DataType item);
}

File: code/poptipwindow/src/main/java/com/bingor/poptipwindow/impl/OnItemSelectListener.kt
package com.bingor.poptipwindow.impl

import com.bingor.poptipwindow.view.wheel.WheelView


/**
 * 滚轮回调
 * [WheelView.setOnItemSelectListener]
 * [WheelView.itemSelectedCallback]
 * Created by HXB on 2018/10/16.
 */
interface OnItemSelectListener<DataType> {
    /**
     * 滑动选择回调
     *
     * @param index 当前选择项的索引
     */
    fun <View : WheelView<*>> onSelected(wheelView: View, index: Int, item: DataType)
}